File: MaxSubArray.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
        
        int Current = nums[0];
        int Output = nums[0];
        
        for (int i = 1; i < nums.Length; i++)
        {
            Current = Math.Max(Current + nums[i], nums[i]);
            
            Output  = Math.Max(Output, Current);
        }
        
        return Output;
    }
}

File: Python/maxSubArray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        
        left = 0
        end = len(nums) - 1 
        maxSum = max(nums)
        
        while (left <= end):

            right = left + 1

            while (right <= end):

                maxSum = max(maxSum, sum(nums[left:right + 1]))

                # print(str(sum(nums[left:right + 1])) + ": " + str(nums[left:right + 1]))

                right += 1

            left += 1

        return maxSum